gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -150,117 +201,300 @@ def test_unauthorized_user_cannot_list_tes_tasks(self):
 
-        response = self.gen3_workflow.get_tes_task(
+        # Step 2: Retrieve the task to check its state
+        task_info = self.gen3_workflow.get_tes_task(
             task_id=task_id,
             user=self.valid_user,
             expected_status=200,
         )
+
         assert (
-            "state" in response
-        ), f"Response must have `state` field in it. But found {response} instead"
+            "state" in task_info
+        ), f"Response must have `state` field in it. But found {task_info} instead"
 
-        response = self.gen3_workflow.cancel_tes_task(
+        # Step 3: Cancel the TES task
+        self.gen3_workflow.cancel_tes_task(
             task_id=task_id,
             user=self.valid_user,
             expected_status=200,
         )
-        assert "state" in response and response["state"] in [
-            "CANCELING",
-            "CANCELED",
-        ], f"Task state should be 'CANCELING', 'CANCELED' after cancellation. But found {response} instead"
+
+        # Step 4: Verify the task is cancelled
+        task_info = self.gen3_workflow.get_tes_task(
+            task_id=task_id,
+            user=self.valid_user,
+            expected_status=200,
+        )
+
+        final_state = task_info.get("state", None)
+        valid_states = {"CANCELING", "CANCELED"}
+        assert (
+            final_state in valid_states
+        ), f"Task state should be one of {valid_states} after cancellation. But found {final_state} instead. Task Info: {task_info}"
+
+    def test_nextflow_workflow(self):
+        """
+        Test Case: Verify that a Nextflow workflow can be executed successfully.
+        """
+        expected_task_outputs = {
+            "extract_metadata (1)": {
+                "filename": "dicom-metadata-img-1.dcm.csv",
+                "command": "python3 /utils/extract_metadata.py img-1.dcm",
+            },
+            "extract_metadata (2)": {
+                "filename": "dicom-metadata-img-2.dcm.csv",
+                "command": "python3 /utils/extract_metadata.py img-2.dcm",
+            },
+            "dicom_to_png (1)": {
+                "filename": "img-1.png",
+                "command": "python3 /utils/dicom_to_png.py img-1.dcm",
+            },
+            "dicom_to_png (2)": {
+                "filename": "img-2.png",
+                "command": "python3 /utils/dicom_to_png.py img-2.dcm",
+            },
+        }
+        workflow_dir = "gen3-integration-tests/test_data/gen3_workflow/"
+        workflow_log = self.gen3_workflow.run_nextflow_task(
+            workflow_dir=workflow_dir,
+            workflow_script="main.nf",
+            nextflow_config_file="nextflow.config",
+        )
+
+        completed_tasks = [
+            self._nextflow_parse_completed_line(line)
+            for line in workflow_log.split("\n")
+            if "Task completed > TaskHandler" in line
+        ]
+
+        for task in completed_tasks:
```

Comment:
i think we should print the contents of the nextflow log file before the `for` loop, so that if an assertion fails, we have more information to debug